feat: add support for utf8view types

NguyenHoangSon96 opened the pull request:
## What's Changed

Please fill in a description of the changes here.

**This contains breaking changes.** <- Remove this line if there are no breaking changes.

Closes #NNN.

NguyenHoangSon96 commented:
> I have not run the code locally, but it looks reasonable to me, and the tests are passing.
Hi @domoritz 
Thank you. Can you merge it?
And if it merged, how long the new version of arrow-js be released

trxcllnt commented:
Maybe I'm missing something about this PR, but this seems like the `Utf8View` just duplicates everything from `Utf8`, and doesn't actually provide a "view" over Utf8 bytes? I assume a real `Utf8View` implementation would include a scalar type that lazily decodes the bytes into a JS string on demand?

If this is just duplicating the Utf8 code, we should just interpret the `Utf8View` typeId as a `Utf8` typeId and reuse all the existing codepaths. We already get complaints about library bundle size, we shouldn't add to it if it can be helped.

domoritz commented:
Good catch and agreed that we should not just duplicate code if the logic is the same.

kou commented:
We can release a new version whenever we release a new version because we split the JS implementation from apache/arrow.
For example, we can release a new version once this is merged.

NguyenHoangSon96 commented:
Hi guys
I closed this branch because I used a force push after I rebased on the main branch.
I moved the branch to here [331](https://github.com/apache/arrow-js/pull/311).
I don't have time to work on this anymore. So if anyone continues to finish this, I will very appreciate it.
Sorry because I can't finish this.
